wrapper/src/libvmaf.h

```diff
@@ -32,92 +34,87 @@
 extern "C" {
 #endif
 
-int compute_vmaf(double* vmaf_score, char* fmt, int width, int height, int (*read_frame)(float *ref_data, float *main_data, float *temp_data, int stride_byte, void *user_data),
-				 void *user_data, char *model_path, char *log_path, char *log_fmt, int disable_clip, int disable_avx, int enable_transform, int phone_model, int do_psnr,
-				 int do_ssim, int do_ms_ssim, char *pool_method, int n_thread, int n_subsample, int enable_conf_interval);
+#include <stdbool.h>
 
-#ifdef __cplusplus
-}
-#endif
+enum VmafLogFmt {
+    VMAF_LOG_FMT_XML  = (1 << 0),
+    VMAF_LOG_FMT_JSON = (1 << 1),
+    VMAF_LOG_FMT_CSV  = (1 << 2),
+};
 
-#ifdef __cplusplus
-#include <vector>
-#include <cstring>
-#include <map>
-#include <memory>
-#include <string>
+enum VmafPoolingMethod {
+    VMAF_POOL_MIN                                      = (1 << 0),
+    VMAF_POOL_MEAN                                     = (1 << 1),
+    VMAF_POOL_HARMONIC_MEAN                            = (1 << 2),
+};
 
-class Asset
+typedef struct VmafFeatureCalculationSetting
 {
-public:
-    Asset(int w, int h, const char *fmt);
-    Asset(int w, int h);
-    int getWidth();
-    int getHeight();
-    const char* getFmt();
-private:
-    const int w, h;
-    const char *fmt;
+    unsigned int n_threads;
+    unsigned int n_subsample;
+    bool disable_avx;
+} VmafFeatureCalculationSetting;
+
+enum VmafFeatureModeSetting {
+    VMAF_FEATURE_MODE_SETTING_DO_NONE                  = (1 << 0),
+    VMAF_FEATURE_MODE_SETTING_DO_PSNR                  = (1 << 1),
+    VMAF_FEATURE_MODE_SETTING_DO_SSIM                  = (1 << 2),
+    VMAF_FEATURE_MODE_SETTING_DO_MS_SSIM               = (1 << 3),
+    VMAF_FEATURE_MODE_SETTING_DO_COLOR                 = (1 << 4),
 };
 
-enum ScoreAggregateMethod
-{
-    MEAN,
-    HARMONIC_MEAN,
-    MINIMUM
+enum VmafPixelFormat {
+    VMAF_PIX_FMT_YUV420P                               = 0,
+    VMAF_PIX_FMT_YUV422P                               = 1,
+    VMAF_PIX_FMT_YUV444P                               = 2,
+    VMAF_PIX_FMT_YUV420P10LE                           = 3,
+    VMAF_PIX_FMT_YUV422P10LE                           = 4,
+    VMAF_PIX_FMT_YUV444P10LE                           = 5,
+    VMAF_PIX_FMT_UNKNOWN                               = 6,
 };
 
-class StatVector
+typedef struct VmafPicture
 {
-public:
-    StatVector();
-    StatVector(std::vector<double> l);
-    std::vector<double> getVector();
-    double mean();
-    double minimum();
-    double harmonic_mean();
-    double second_moment();
-    double percentile(double perc);
-    double var();
-    double std();
-    void append(double e);
-    double at(size_t idx);
-    size_t size();
-private:
-    std::vector<double> l;
-    void _assert_size();
-};
+    float *data[3];
+    unsigned int w[3], h[3];
```

Comment:
No need to store the width and height of all planes, it should be derived from pix_fmt. Just use `unsigned w, h`.

Reply:
The reason I did this is to do the derivation of width and height of all planes in one place (e.g. in read_frame or whoever populates/allocates a VmafPicture). Otherwise you have to go and do the calculation in every feature extractor that is applied on color.